cranelift-codegen/meta/src/cdsl/instructions.rs

```diff
@@ -1078,6 +1086,16 @@ fn bind(
         }
     }
 
+    verify_polymorphic_binding(&inst, &value_types);
+
+    BoundInstruction { inst, value_types }
+}
+
+/// Helper to verify that binding types to the instruction does not violate polymorphic rules
+fn verify_polymorphic_binding(
+    inst: &Instruction,
+    value_types: &Vec<ValueTypeOrAny>,
+) {
```

Comment:
This just factors out some verification that I thought I should re-use above in `bind_vector`

Reply:
:+1: